seqr/views/utils/variant_utils.py

```diff
@@ -0,0 +1,61 @@
+import json
+import logging
+from django.contrib.auth.models import User
+
+from seqr.models import SavedVariant
+from xbrowse_server.api.utils import add_extra_info_to_variants_project
+from xbrowse_server.mall import get_reference
+from xbrowse_server.base.models import Project as BaseProject
+from xbrowse_server.base.lookups import get_variants_from_variant_tuples
+
+
+def get_or_create_saved_variant(xpos=None, ref=None, alt=None, family=None, project=None, **kwargs):
+    if not project:
+        project = family.project
+    saved_variant, _ = SavedVariant.objects.get_or_create(
+        xpos_start=xpos,
+        xpos_end=xpos + len(ref) - 1,
```

Comment:
in retrospect, since xpos_end is redundant in this case, how about we get rid of xpos_start/end and just have xpos

Reply:
I like that idea, but thats out of the scope of this particular issue since it would require a bunch of updates unrelated files and a migration. Can you make a separate ticket?